### ISSUE TITLE
Add compression level

### ISSUE THREAD

**Issue by Noxmiles** [nextcloud/files_zip#33]
At the moment the files are only saved in a zip, but not compressed. You could add a compression level in the settings.

I'm not sure, do you use this? That has compression available. https://github.com/McNetic/PHPZipStreamer/blob/master/MANUAL.md

**Comment by juliushaertl** [comment]
Yes, a fork of it which is part of the Nextcloud server 3rdparty libraries: https://github.com/DeepDiver1975/PHPZipStreamer

Adding compression would make sense

**Comment by Noxmiles** [comment]
Little reminder, maybe..? :)

**Comment by StarSmasher44** [comment]
Could be somewhat useful :)

**Comment by juliushaertl** [comment]
Pull requests are very welcome of course 😉

**Comment by StarSmasher44** [comment]
Looked at it for a bit, unfortunately while I can program relatively Ok.... Arrays scare me.
So other than not knowing what to put where in the phpzipstreamer call, I *could* do it.

But as I have fairly limited time don't expect much :P

**Comment by joshtrichards** [comment]
I looked into this elsewhere earlier in the week: https://github.com/nextcloud/server/issues/38456#issuecomment-1563715552

**Comment by tcitworld** [comment]
It looks like https://github.com/maennchen/ZipStream-PHP could be a nice replacement

**Comment by opus-nbonenberger** [comment]
@juliusknorr A zip extension without compression somehow defeats its purpose. Can you add compression, please?
